packages/frontend-v2/hooks/useFetchSearchCourses.ts

```diff
@@ -0,0 +1,50 @@
+import useSWR, {SWRResponse} from "swr";
+import { SearchAPI } from "@graduate/api-client";
+import { AxiosError } from "axios";
+import { ScheduleCourse2 } from "@graduate/common";
+
+type FetchCoursesResponse = Omit<SWRResponse<ScheduleCourse2<null>[], AxiosError | Error>, "data" | "mutate">
+
+type FetchCoursesReturn = FetchCoursesResponse & {
+    courses? : ScheduleCourse2<null>[];
+    isLoading : boolean;
+}
+
+/**
+ * 
+ * @param subject - the type of class that we are fetching from SearchNEU
+ * @param classId - the identification number of the class as a string
+ */
+export function useFetchSearchCourses(courses: {subject: string, classId: string}[]): FetchCoursesReturn{
```

Comment:
Super nit: I would just type this out separately like `type course = {...}` and then `courses: course[]`